### PR TITLE
Issue 415 scope on type

### LINKED ISSUE #415: Scoping context based on the @type
Currently, for JSON data we have, I have to put `@context`s all around the JSON to scope properties:

``` json
{
    "@context": {
        "@base": "http://127.0.0.1:8000/api/v2/node/",
        "@vocab": "http://127.0.0.1:8000/api/v2/node/?format=vocab#"
    },
    "@type": "Node",
    "@id": "000c3d9d-5d7c-492a-8d3b-bb8b922ae35f",
    "config": {
        "core.location": {
            "@context": {
                "@base": "_:config/core.location/",
                "@vocab": "http://127.0.0.1:8000/api/v2/node/?format=vocab#config/core.location/"
            },
            "@id": "123",
            "@type": "Location",
            "timezone": "Europe/Ljubljana",
            "address": "San Juan, Puerto Rico",
            "city": "Ljubljana",
            "country": "SI",
            "geolocation": {
                "@context": "http://geojson.org/contexts/geojson-base.jsonld",
                "type": "Point",
                "coordinates": [
                    -66.073756,
                    18.378287
                ]
            },
            "altitude": 0.0
        },
        "core.project": {
            "@context": {
                "@base": "_:config/core.project/",
                "@vocab": "http://127.0.0.1:8000/api/v2/node/?format=vocab#config/core.project/"
            },
            "@id": "444",
            "@type": "Project",
            "project": "http://127.0.0.1:8000/api/v2/project/2"
        }
    }
}
```

It would be great if I could somehow `@vocab` (or `@context`) itself would be scoped based on the value of `@type`, similarly to what I have to do currently manually. Ideally, JSON-LD would then be something like:

``` json
{
    "@context": {
        "@base": "http://127.0.0.1:8000/api/v2/node/",
        "@vocab": "http://127.0.0.1:8000/api/v2/node/?format=vocab#"
    },
    "@type": "Node",
    "@id": "000c3d9d-5d7c-492a-8d3b-bb8b922ae35f",
    "config": {
        "core.location": {
            "@id": "_:config/core.location/123",
            "@type": "config/core.location/Location",
            "timezone": "Europe/Ljubljana",
            "address": "San Juan, Puerto Rico",
            "city": "Ljubljana",
            "country": "SI",
            "geolocation": {
                "@context": "http://geojson.org/contexts/geojson-base.jsonld",
                "type": "Point",
                "coordinates": [
                    -66.073756,
                    18.378287
                ]
            },
            "altitude": 0.0
        },
        "core.project": {
            "@id": "_:config/core.project/444",
            "@type": "config/core.project/Project",
            "project": "http://127.0.0.1:8000/api/v2/project/2"
        }
    }
}
```

And all definitions of properties would stay exactly the same as in the initial JSON at the top.

### PR DESCRIPTION
**Pull request by gkellogg**
This commit adds scoping based on `@type` for expansion and compaction. Also ensure that types used in type maps have any associated local context applied to their values.

Also, corrects some minor errors in id and type maps where values were not appropriately expanded ore compacted.

Fixes #415.

### CODE REVIEW

File: spec/latest/json-ld-connect/index.html [hunk 1]
```diff
@@ -194,7 +194,7 @@
     <dd>An unordered labeled directed graph, where <tref title="node">nodes</tref> are
       <tref title="IRI">IRIs</tref> or <tref title="Blank Node">Blank Nodes</tref>, or other values.
       A <tref>JSON-LD graph</tref> is a generalized representation of a
-      <cite><a href="http://www.w3.org/TR/rdf11-concepts/#dfn-rdf-graph">RDF graph</a></cite>
+      <cite><a href="http://www.w3.org/TR/RDF11-CONCEPTS/#dfn-rdf-graph">RDF graph</a></cite>
```
Comment (lanthaler) [inline spec/latest/json-ld-connect/index.html]: This URL redirects to the lower-case version, why did you change it?

**Comment by gkellogg** [comment]
Because we use upper case within `[[]]`, it was creating two references, one normative and one informative. When I discussed this with Melvin, he felt this was appropriate behavior. To get back to a single entry in the reference section, it was necessary to unify the use within `[[]]` and `@data-cite`.

However, I didn't intend to update the json-ld-connect spec, so I'll reverse that change.

**Comment by lanthaler** [comment]
> Because we use upper case within [[]], it was creating two references, one normative and one informative. When I discussed this with Melvin, he felt this was appropriate behavior.

Who is we? ReSpec? That sounds like a bug to me.

**Comment by gkellogg** [comment]
_We_ is, we the editors of JSON-LD. For example, we use `[[RDF11-CONCEPTS]]`, instead of `[[rdf11-concepts]]`, and so-forth. We could, of course, change to use lower-case references uniformly.

It doesn't really matter to me, but to use `@data-cite`, we need to be consistent, and this seems like a pretty nice feature of ReSpec to me.

The fact that ReSpec is case-sensitive seems like a design decision, which I wasn't interested in pursuing. Feel free to take it up on spec-prod.

**Comment by gkellogg** [comment]
@lanthaler do you have an substantive feedback on this PR?

**Comment by lanthaler** [comment]
No, and unfortunately I don't have the time to properly review it. As discussed elsewhere I'm concerned about the complexity of JSON-LD but see the value of this feature. So I'm fine with this being merged.
